.github/workflows/build.yml

```diff
@@ -12,16 +12,24 @@ jobs:
 
     runs-on: ubuntu-20.04
     container: espressif/idf:${{ matrix.idf_ver }}
-    steps:
-      - name: Checkout esp_eth_drivers
-        uses: actions/checkout@master
-        with:
-          path: esp_eth_drivers
-      - name: Build ${{ matrix.example }} with IDF-${{ matrix.idf_ver }} for ${{ matrix.idf_target }}
-        env:
-          IDF_TARGET: ${{ matrix.idf_target }}
-        shell: bash
-        run: |
-          . ${IDF_PATH}/export.sh
-          cd $GITHUB_WORKSPACE/esp_eth_drivers/ksz8863/examples/${{ matrix.example }}
-          idf.py build
+      steps:
+        - name: Checkout esp_eth_drivers
+          uses: actions/checkout@master
+          with:
+            path: esp_eth_drivers
+        - name: Build ${{ matrix.example }} with IDF-${{ matrix.idf_ver }} for ${{ matrix.idf_target }}
+          env:
+            IDF_TARGET: ${{ matrix.idf_target }}
+          shell: bash
+          run: |
+            . ${IDF_PATH}/export.sh
+            cd $GITHUB_WORKSPACE/esp_eth_drivers/ksz8863/examples/${{ matrix.example }}
+            idf.py build
+        - name: Build lan867x/lan867x_tcp_sockets with IDF-${{ matrix.idf_ver }} for ${{ matrix.idf_target }}
```

Comment:
Why not merge the build in a single step and use the matrix to set the paths?
You might consider store the `example` element as array map.